### ISSUE TITLE
Detection delay

### ISSUE THREAD

**Issue by DocMAX** [xbmc/peripheral.joystick#137]
My Xbox Controller is detected by Kodi with a delay of about 5 sec. With the xboxdrv driver there is no delay and the controller works imedatelly after connection/Kodi start.

**Comment by garbear** [comment]
Wireless controller, right? If you push a button it should detect immediately, though the first button press won't have any effect. This is a problem with the joystick driver.

**Comment by DocMAX** [comment]
Yes wireless. I have to push a button and then wait a few seconds. I dont have this effect with xboxdrv, but with xpad driver.

**Comment by garbear** [comment]
When the wireless receiver is connected, xpad shows 4 joysticks, even though one 1 is connected. (This might be fixed in recent kernels). To solve this, wireless Xbox controllers are only detected once the first button is pressed. However, once you press a button the controller should be detected immediately, and the second button press should work right away. This part seems to be broken, so just in case I added a failsafe scan every 5 seconds.

I'll look into why the first button press doesn't activate the controller immediately. If xpad has been fixed, then I'll remove this workaround completely.
